src/OSPSuite.Presentation/Presenters/Importer/UnitsEditorPresenter.cs

```diff
@@ -18,7 +18,7 @@ public class UnitsEditorPresenter : AbstractDisposablePresenter<IUnitsEditorView
 
       private bool _columnMapping;
 
-      public UnitDescription Unit => new UnitDescription(_selectedUnit, _selectedColumn);
+      public UnitDescription Unit => new UnitDescription(_view.SelectedUnit, _selectedColumn);
```

Comment:
Why not use _selectedUnit. Why are they out of sync? This smells. Remove the _selectedUnit if it's not used